src/sparse/highlevel.jl

```diff
@@ -0,0 +1,30 @@
+(\)(A::AbstractTriangular{<:CuSparseMatrix},B::CuMatrix)       = sm('N',A,B,'O')
+(\)(transA::Transpose{<:Any, AbstractTriangular{<:CuSparseMatrix}}, B::CuMatrix) = sm('T',transA.parent,B,'O')
+(\)(adjA::Adjoint{<:Any, AbstractTriangular{<:CuSparseMatrix}},B::CuMatrix) = sm('C',adjA.parent,B,'O')
+
+(*)(A::CuSparseMatrix,B::CuVector)       = mv('N',A,B,'O')
```

Comment:
What's the reason you chose to implement the non-mutating versions, instead of directly using `mv!` and `mm2!`?

Reply:
I'm not entirely sure either, but I figured implementing the in-place ops is more versatile since there's asbtract functionality in Base for allocating containers with non-mutating ops, eg. https://github.com/JuliaLang/julia/blob/eabc5de03131e780a8be3b58dd576f1007b9ce99/stdlib/LinearAlgebra/src/matmul.jl#L140-L143